File: src/components/JSONHandler/src/ALRPCObjectsImpl/V2/VehicleDataResult.cpp
#include "../include/JSONHandler/ALRPCObjects/V2/VehicleDataResult.h"
#include "VehicleDataResultMarshaller.h"
#include "VehicleDataResultCodeMarshaller.h"
#include "VehicleDataTypeMarshaller.h"
/*
  interface	Ford Sync RAPI
  version	2.0O
  date		2012-11-02
  generated at	Thu Dec 13 14:18:29 2012
  source stamp	Thu Dec 13 14:18:27 2012
  author	robok0der
*/



using namespace NsAppLinkRPCV2;

VehicleDataResult::VehicleDataResult(const VehicleDataResult& c)
{
  *this=c;
}


bool VehicleDataResult::checkIntegrity(void)
{
  return VehicleDataResultMarshaller::checkIntegrity(*this);
}


VehicleDataResult::VehicleDataResult(void)
{
}



bool VehicleDataResult::set_dataType(const VehicleDataType& dataType_)
{
  if(!VehicleDataTypeMarshaller::checkIntegrityConst(dataType_))   return false;
  dataType=dataType_;
  return true;
}

bool VehicleDataResult::set_resultCode(const VehicleDataResultCode& resultCode_)
{
  if(!VehicleDataResultCodeMarshaller::checkIntegrityConst(resultCode_))   return false;
  resultCode=resultCode_;
  return true;
}




const VehicleDataType& VehicleDataResult::get_dataType(void) const 
{
  return dataType;
}


const VehicleDataResultCode& VehicleDataResult::get_resultCode(void) const 
{
  return resultCode;
}

File: mobile/android/SyncProxyAndroid/src/com/ford/syncV4/proxy/rpc/VehicleDataResult.java
package com.ford.syncV4.proxy.rpc;

import java.util.Hashtable;

import com.ford.syncV4.proxy.RPCStruct;
import com.ford.syncV4.proxy.constants.Names;
import com.ford.syncV4.proxy.rpc.enums.VehicleDataResultCode;
import com.ford.syncV4.proxy.rpc.enums.VehicleDataType;

public class VehicleDataResult extends RPCStruct {

    public VehicleDataResult() { }
    public VehicleDataResult(Hashtable hash) {
        super(hash);
    }
    public void setDataType(VehicleDataType dataType) {
    	if (dataType != null) {
    		store.put(Names.dataType, dataType);
    	} else {
    		store.remove(Names.dataType);
    	}
    }
    public VehicleDataType getDataType() {
    	return (VehicleDataType) store.get(Names.dataType);
    }
    public void setResultCode(VehicleDataResultCode resultCode) {
    	if (resultCode != null) {
    		store.put(Names.resultCode, resultCode);
    	} else {
    		store.remove(Names.resultCode);
    	}
    }
    public VehicleDataResultCode getResultCode() {
    	return (VehicleDataResultCode) store.get(Names.resultCode);
    }
}
